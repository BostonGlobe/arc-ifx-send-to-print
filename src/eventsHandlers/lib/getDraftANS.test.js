const getDraftANS = require('./getDraftANS');

describe('getDraftANS', () => {
  it('fetches the draft ANS from the Draft API', async () => {
    // This does require that the photo not be modified on Arc, but it'll do for now. I don't feel
    // like mocking Axios at the moment.
    const result = await getDraftANS('VPQMNJPIRNGKHF5GISP5UVLMNA');

    expect(result).toMatchSnapshot();
  });
});
