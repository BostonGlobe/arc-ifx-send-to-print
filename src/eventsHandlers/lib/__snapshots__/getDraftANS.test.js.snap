// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDraftANS fetches the draft ANS from the Draft API 1`] = `
{
  "additional_properties": {
    "clipboard": {},
    "has_published_copy": false,
  },
  "address": {},
  "canonical_website": "bostonglobe",
  "content_elements": [
    {
      "_id": "VTMWDVS4GBEZXEFCU6E535TDYQ",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer non luctus massa, sit amet mollis ipsum. Praesent congue lectus elit, ac rutrum nisl consequat eget. In hac habitasse platea dictumst. In at est rutrum mi convallis elementum.",
      "type": "text",
    },
    {
      "_id": "AYSQREJWOVEXFBF3LQQ2A7HQPQ",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Praesent magna massa, auctor in massa nec, efficitur commodo felis. In non nunc ac nulla ornare laoreet ut a nibh. Nam in imperdiet neque. Pellentesque facilisis tellus non turpis pretium, et elementum neque semper. Proin eleifend justo vel mi commodo sagittis. Pellentesque suscipit ullamcorper neque, porta semper leo. Nam placerat metus vel dolor vehicula convallis.",
      "type": "text",
    },
    {
      "_id": "GKBJZ5MUYNDNJAWY53G5H7LLGM",
      "referent": {
        "id": "GKBJZ5MUYNDNJAWY53G5H7LLGM",
        "provider": "",
        "referent_properties": {
          "additional_properties": {
            "_id": "46XWOZHQ25F3BL5RYYUTP2WNFA",
            "comments": [],
          },
        },
        "type": "image",
      },
      "type": "reference",
    },
    {
      "_id": "UVJONA2K6FGUBNUZNX3VH7HDOQ",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Fusce condimentum, mi vel maximus maximus, diam lorem euismod sem, id porta velit ligula vitae mi. Nulla mattis purus quam, sed consequat dolor ornare vel. Aliquam velit urna, facilisis quis est sed, egestas feugiat arcu. In vel accumsan turpis. Nam commodo aliquet felis, consequat congue tellus ornare at. Vivamus risus lacus, pulvinar eu massa eu, hendrerit ullamcorper augue.",
      "type": "text",
    },
    {
      "_id": "RKUGOX5CCBGFDM4N3QY3LSF55E",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Donec accumsan, risus sit amet tincidunt porta, lacus eros fermentum velit, vitae maximus nisi orci vitae nisl. Ut tellus nisl, feugiat nec ligula sed, convallis laoreet nunc. Integer et augue mauris. Mauris tempus nunc id purus egestas dignissim. Maecenas eu venenatis tortor, nec dictum odio. Duis metus tellus, dictum in nulla eu, aliquam maximus massa.",
      "type": "text",
    },
    {
      "_id": "7FCJV6YIW5DRFI6IRJGKUGKAUE",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Duis pellentesque ullamcorper nisl sit amet viverra. Donec tempus elementum maximus. Duis in imperdiet lectus. Donec ornare ullamcorper commodo. Integer mollis lacus sem, ac viverra arcu tincidunt in. Phasellus congue suscipit dictum. Suspendisse at odio dui. Donec rhoncus lacinia ultrices. Etiam sodales consequat varius. Proin non semper odio, vel pharetra nunc.",
      "type": "text",
    },
    {
      "_id": "66SWLZWNUREYNGBGI6KXFUB5LY",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Sed porttitor tellus sed imperdiet placerat. Morbi venenatis est vel dolor tristique, nec bibendum nisl malesuada. Duis justo tortor, maximus in metus id, commodo semper diam. Vestibulum bibendum porta lectus. Praesent faucibus egestas erat, malesuada ullamcorper quam cursus nec. Morbi tincidunt dui eu dui porta, eu dictum nunc imperdiet.",
      "type": "text",
    },
    {
      "_id": "XYHU7ZYD6RARVJ3R67LLC4D4WU",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Quisque volutpat nulla et molestie feugiat. Maecenas lobortis urna odio, vitae eleifend est vehicula non. Nunc vitae interdum lacus. Suspendisse non nisi at est lobortis lacinia. Vivamus tincidunt diam at sem tempor, sit amet aliquam justo tempor. Morbi aliquam massa sit amet placerat bibendum. Proin eu augue ut nulla mollis bibendum. Mauris ornare ac tortor sed luctus. Praesent lacus ipsum, facilisis quis nibh nec, semper vulputate justo. Sed consectetur vestibulum est, a vestibulum mauris maximus ultrices.",
      "type": "text",
    },
    {
      "_id": "SWTUDZL5TRAZBLXNDUGCRDXFLQ",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Morbi cursus ultricies arcu at malesuada. Curabitur venenatis lectus a mauris sagittis ultrices. Aenean posuere maximus lacus et euismod. Nam rutrum ante at cursus feugiat. Suspendisse feugiat dapibus turpis, vitae ultrices ligula eleifend sed. Suspendisse potenti. Quisque dictum varius diam ac sollicitudin.",
      "type": "text",
    },
    {
      "_id": "SB5TTPJ6KFCIJBJUPZHJYSMM6M",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Nunc nec neque maximus, lobortis dolor sit amet, tristique massa. Proin semper nisi nisl, ut pharetra orci lobortis quis. Praesent pretium odio quis massa lobortis ultricies. Proin id tortor a massa mollis ornare. Aenean eu justo quis nulla fringilla aliquam. Quisque libero leo, pharetra quis imperdiet non, dictum non ex. Morbi dignissim accumsan dui. Aliquam in auctor est, a pulvinar felis. Aenean congue ligula lacinia dolor lobortis ullamcorper.",
      "type": "text",
    },
    {
      "_id": "XEUGGQZGQ5C7JFJ27G5SXGDOPA",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Nunc imperdiet, elit sed lobortis iaculis, mi lectus aliquet diam, ac semper dui est ut justo. Phasellus malesuada elit id vulputate sagittis. Etiam bibendum lectus sapien, vel elementum ipsum interdum non. Morbi sed aliquam felis. Nullam ac efficitur tellus. Morbi purus elit, lacinia sed purus vitae, eleifend mattis quam. Mauris non faucibus ante. Fusce ac egestas tellus. Sed hendrerit elit nunc, vel condimentum lacus ultricies id. Curabitur odio turpis, ultricies non purus vitae, mattis ultricies lacus. Mauris a sem ultrices, fringilla elit id, viverra orci. Pellentesque laoreet convallis dui, in fermentum nisl dignissim eu. Pellentesque justo nunc, vulputate eu nibh eget, euismod tempor nulla. Nullam ut malesuada augue.",
      "type": "text",
    },
    {
      "_id": "VQLY2US4H5BSHP5AF5UG5USYKU",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Curabitur imperdiet quam vitae lobortis consectetur. Ut scelerisque eros a neque scelerisque, tristique hendrerit felis euismod. Sed accumsan velit eu ullamcorper fringilla. Aliquam luctus ut dui vel fermentum. Duis erat purus, scelerisque id convallis vitae, luctus quis risus. Donec at aliquet justo. Donec ornare rutrum velit, non tincidunt ex lacinia eu. Proin blandit sagittis leo nec interdum. Nulla rutrum pharetra enim, vitae tristique ligula rhoncus et.",
      "type": "text",
    },
    {
      "_id": "CBTZHJ3W2JEDREIRTG4QRQF6HI",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Proin nibh ex, porttitor eu facilisis nec, bibendum consequat augue. Vivamus malesuada, orci vel mattis varius, erat sapien finibus lacus, id tempus enim nulla vel diam. Nulla non lacus ipsum. Cras ultricies sed elit vel pretium. Curabitur viverra malesuada bibendum. Aenean consequat, nibh sed euismod condimentum, libero ex fringilla neque, ac imperdiet elit lectus sed arcu. Vestibulum ultrices ultricies nisi a semper. Praesent nibh justo, tempus non venenatis a, aliquam ac velit. Maecenas sed ex eu mi rhoncus facilisis. Ut ultricies ligula sed felis eleifend posuere. Sed nisi ipsum, consectetur ac ligula vel, consequat tristique sapien.",
      "type": "text",
    },
    {
      "_id": "QDZIAOBMWVHTZKH5JVPSCX5ZRI",
      "additional_properties": {
        "comments": [],
        "inline_comments": [],
      },
      "content": "Nam tincidunt sapien in mauris pellentesque tempus. In tincidunt sem ac tellus maximus, vel condimentum enim dictum. Vivamus blandit commodo ligula, id laoreet enim feugiat nec. Aenean sed risus tellus. Praesent sed urna sed nisl eleifend dictum a vitae ante. Vivamus id pretium metus, sit amet maximus magna. Vivamus maximus pellentesque dolor, eget sagittis sapien placerat et. Nam malesuada nulla quis quam elementum vestibulum in eu leo. Donec egestas interdum vulputate. Mauris sit amet euismod nulla. Praesent ac sollicitudin nisl. Pellentesque in enim est. Sed vitae augue consectetur, tincidunt lacus sit amet, posuere leo. Curabitur leo lacus, ullamcorper quis tortor ut, gravida pellentesque nunc. Duis commodo eros et dictum aliquam.",
      "type": "text",
    },
  ],
  "content_restrictions": {
    "content_code": "non-exempt",
  },
  "created_date": "2024-11-20T21:21:36.121Z",
  "credits": {
    "by": [
      {
        "referent": {
          "id": "todd.dukart",
          "provider": "",
          "referent_properties": {},
          "type": "author",
        },
        "type": "reference",
      },
    ],
  },
  "description": {
    "basic": "A test article for automated Send to Print testing.",
  },
  "distributor": {
    "category": "staff",
    "name": "globestaff",
    "subcategory": "",
  },
  "headlines": {
    "basic": "Toddâ€™s STP Automation Test - DO NOT MODIFY",
    "meta_title": "",
    "mobile": "",
    "native": "",
    "print": "",
    "tablet": "",
    "web": "",
  },
  "label": {},
  "language": "",
  "last_updated_date": "2024-11-20T22:01:38.709Z",
  "owner": {
    "id": "sandbox.bostonglobe",
    "sponsored": false,
  },
  "planning": {
    "internal_note": "",
    "story_length": {
      "character_count_actual": 5905,
      "character_encoding": "UTF-16",
      "word_count_actual": 871,
    },
  },
  "promo_items": {
    "basic": {
      "_id": "PUWLWI4TH5EMNPZSVZPLLJT5D4",
      "referent": {
        "id": "PUWLWI4TH5EMNPZSVZPLLJT5D4",
        "provider": "",
        "referent_properties": {},
        "type": "image",
      },
      "type": "reference",
    },
  },
  "related_content": {
    "basic": [],
    "redirect": [],
  },
  "revision": {
    "branch": "default",
    "parent_id": "MU4PBRSPLZE6XO4N3NB53C26WY",
    "published": false,
    "revision_id": "SOFYETQSCFFGNBHBRHUNJPOFRA",
    "user_id": "todd.dukart@globe.com",
  },
  "slug": "xxautomationtest",
  "source": {
    "name": "globestaff",
    "source_type": "staff",
    "system": "composer",
  },
  "subheadlines": {
    "basic": "",
  },
  "type": "story",
  "version": "0.10.10",
  "workflow": {
    "status_code": 1,
  },
}
`;
